source/includes/usage-examples/retrieve-code-examples.c

```diff
@@ -0,0 +1,96 @@
+#include <bson/bson.h>
+#include <mongoc/mongoc.h>
+#include <stdio.h>
+
+int
+main (int argc, char *argv[])
+{
+    mongoc_client_t *client;
+    mongoc_collection_t *collection;
+    mongoc_cursor_t *results;
+    int64_t count;
+    const bson_t *doc;
+    bson_t *query, *command;
```

Comment:
Not all examples use `query` or `command`. Running an example may call `bson_destroy` on an uninitialized `query`. Example with the "Retrieve Distinct Values":

```c
#include <bson/bson.h>
#include <mongoc/mongoc.h>
#include <stdio.h>

int main(int argc, char *argv[])
{
    mongoc_client_t *client;
    mongoc_collection_t *collection;
    mongoc_cursor_t *results;
    int64_t count;
    const bson_t *doc;
    bson_t *query, *command;
    bson_t reply;
    char *str;
    bson_error_t error;

    mongoc_init();

    client =
        mongoc_client_new("<connection string URI>");
    collection =
        mongoc_client_get_collection(client, "<database name>", "collection name");

    // Start example code here
    command = BCON_NEW ("distinct",
                        BCON_UTF8 ("<collection name>"),
                        "key",
                        BCON_UTF8 ("<field name>"));

    if (!mongoc_collection_read_command_with_opts (collection, command, NULL, NULL, &reply, &error)) {
        fprintf (stderr, "An error occurred: %s\n", error.message);
    } else {
        str = bson_as_canonical_extended_json (&reply, NULL);
        printf ("%s\n", str);
        bson_free (str);
    }
    // End example code here

    bson_destroy(query); // query is unintialized!
    mongoc_collection_destroy(collection);
    mongoc_client_destroy(client);
    mongoc_cleanup();

    return EXIT_SUCCESS;
}
```

I recommend revising examples to create a new scope to isolate variables needed for the example. Examples:
```c
// start-find
{
    bson_t *query = bson_new();
    // Add fields to `query` if needed.
    mongoc_cursor_t *results = mongoc_collection_find_with_opts(collection, query, NULL, NULL);
    while (mongoc_cursor_next(results, &doc))
    {
        char *str = bson_as_canonical_extended_json(doc, NULL);
        printf("%s\n", str);
        bson_free(str);
    }
    mongoc_cursor_destroy(results);
    bson_destroy(query);
}
// end-find

// start-distinct
{
    bson_t reply;
    bson_t *command = BCON_NEW("distinct",
                                BCON_UTF8("<collection name>"),
                                "key",
                                BCON_UTF8("<field name>"));

    if (!mongoc_collection_read_command_with_opts(collection, command, NULL, NULL, &reply, &error))
    {
        fprintf(stderr, "An error occurred: %s\n", error.message);
    }
    else
    {
        char *str = bson_as_canonical_extended_json(&reply, NULL);
        printf("%s\n", str);
        bson_free(str);
    }
    bson_destroy(&reply);
    bson_destroy(command);
}
// end-distinct
```